docs/02-app/01-building-your-application/04-caching/index.mdx

```diff
@@ -0,0 +1,468 @@
+
+```jsx
+// Opt out of caching
+fetch(`https://...`, { cache: 'no-store' })
+```
+
+Alternatively, you can also use the [Route Segment Config options](#segment-config-options) to opt out of caching for **all** data requests in a route segment.
+
+> **Vercel Data Cache**
+>
+> If your Next.js application is deployed to Vercel, we recommend reading the [Vercel Data Cache](https://vercel.com/docs/infrastructure/data-cache) documentation for a better understanding of Vercel specific features.
+
+### Full Route Cache
+
+Next.js automatically caches the rendered result of React Components on the server at build time. This is an optimization that allows you to skip rendering components on the server on every request, and instead directly send the cached result to the client, resulting in faster page loads.
+
+> **Related terms**:
+>
+> - The terms **Automatic Static Optimization**, **Static Site Generation**, or **Static Rendering** might be also be used interchangeably to refer to the process of rendering and caching parts of your application code on the server at build time.
+
+To understand how the Full Route Cache works, it's helpful to look at how React handles rendering, and how Next.js caches the result:
+
+#### 1. React Rendering on the Server
+
+On the server, Next.js uses React's APIs to orchestrate rendering. This process is split into three stages:
+
+1. React renders Server Components into a special data format, optimized for streaming, called the **React Server Component Payload**.
+2. Next.js compiles, bundles, and code-splits the **JavaScript instructions** for rendering Client Components on the client.
+3. Next.js uses the React Server Component Payload and JavaScript instructions to pre-render **HTML** on the server.
+
+> **What is React Server Component Payload?**
+>
+> The React Server Component Payload is a compact string representation of the React Component tree. It's used by React on the client to update the DOM. The React Server Component Payload contains:
+>
+> - The rendered result of Server Components
+> - Placeholders for where Client Components should be rendered on the client
+> - Any serialized props passed to these Client Components
+> - References to the JavaScript instructions needed to render the Client Components
+> - Metadata tags, such as references to stylesheets
+>
+> To learn more, see the [React Rendering](/docs/app/building-your-application/rendering/react-components) documentation.
+
+#### 2. Next.js Caching on the Server (Full Route Cache)
+
+On the server, at build time, Next.js caches the React Server Component Payload, Client Component JavaScript, and rendered HTML.
+
+#### 3. Next.js Caching on the Client (Router Cache)
+
+At request time, the React Server Component Payload is stored in the client-side [Router Cache](#router-cache) - a separate in-memory cache, split by individual routes. This Router Cache is used to improve the navigation experience by prefetching.
```

Comment:
For dynamic segments (`app/blog/[slug]/page.js` for example) it can also cache in the Full Route Cache at request time for paths that weren't known at build time, where generateStaticParams is essentially a seed for the cache, not all entries. This can be configured by setting `export const dynamicParams = false`. When that flag is added only the paths in generateStaticParams will be served, the others 404 / match other routes (i.e. when you have a catch-all route).

Reply:
Nice, good to know. I'll add that as a note.